python/nvforest/nvforest/benchmark/run.py

```
@@ -0,0 +1,856 @@
+
+
+def _check_lightgbm_gpu_available() -> bool:
+    """Check if LightGBM was built with GPU support."""
+    try:
+        # Try to create a dummy dataset and check if GPU device works
+        # This is a lightweight check that doesn't require actual training
+        params = {"device": "gpu", "gpu_platform_id": 0, "gpu_device_id": 0}
+        _ = lgb.Dataset([], params=params)
+        # LightGBM will raise an error if GPU is not available when we try to use it
+        return True  # Assume available, will fail gracefully during training/predict
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)